yarl/_url.py

```diff
@@ -1199,7 +1218,45 @@ def join(self, url: "URL") -> "URL":
         # See docs for urllib.parse.urljoin
         if not isinstance(url, URL):
             raise TypeError("url should be URL")
-        return URL(urljoin(str(self), str(url)), encoded=True)
+        other: URL = url
```

Comment:
Note to self: do performance testing with this change

Reply:
I'm preparing an RC release to make sure the added typing doesn't cause any problems.  Once thats done I'll try to do some production testing with this tonight if I don't run out of waking hours.